### PR TITLE
[TEST] Improve unit test coverage for apiserver `pkg/model`

### PR DESCRIPTION
**Pull request by JiangJiaWei1103**
## Changes
1. Fix linting issues
2. Add a test case for `nil` input in `convertAutoscalingOptions`
3. Add a test for the `ImagePullPolicy` field in `PopulateWorkerNodeSpec`
4. Add unit tests for functions that convert slices of K8s-native CRD objects into KubeRay’s internal API representations:
    * `FromCrdToAPIClusters`
    * `FromKubeToAPIComputeTemplates`
    * `FromCrdToAPIJobs`

## Why are these changes needed?
To ensure high confidence in code quality before releasing the APIServer to the public, we aim to improve unit test coverage. Currently, the `pkg/model` package within the APIServer has low unit test coverage, as shown below:

### Before
Screenshot 2025-04-18 at 12 07 54 AM

### After
After adding new tests above, the unit test coverage boosts from 74.7% to 79.3%.
Screenshot 2025-04-19 at 12 22 55 PM

## Related issue number
https://github.com/ray-project/kuberay/issues/3316

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
  - [x] Unit tests
  - [ ] Manual tests
  - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by JiangJiaWei1103** [comment]
Hi @dentiny,

I’ve added unit tests to cover several previously untested functions. To avoid overloading this PR with too many changes, would it be okay if I stop here and open a follow-up PR to continue extending the remaining test cases?

Thanks!

**Comment by dentiny** [comment]
> Hi @dentiny,
> 
> I’ve added unit tests to cover several previously untested functions. To avoid overloading this PR with too many changes, would it be okay if I stop here and open a follow-up PR to continue extending the remaining test cases?
> 
> Thanks!

That totally makes sense, thank you for the help!

File: apiserver/pkg/model/converter_test.go [hunk 1]
```diff
@@ -568,6 +572,21 @@ func TestAutoscalerOptions(t *testing.T) {
 	assert.Len(t, options.Volumes, 2)
 }
 
+func TestNilAutoscalerOptions(t *testing.T) {
+	options := convertAutoscalingOptions(nil)
+	assert.Nil(t, options)
+}
+
+func TestFromCrdToAPIClusters(t *testing.T) {
```
Comment (dentiny) [inline apiserver/pkg/model/converter_test.go:580]: Could you please help pre-allocate here?
https://github.com/ray-project/kuberay/blob/747708b7cc123eb270ada8f84dbc4a988836db03/apiserver/pkg/model/converter.go#L92

File: apiserver/pkg/model/converter_test.go [hunk 2]
```diff
@@ -642,6 +670,16 @@ func tolerationToString(toleration *api.PodToleration) string {
 	return "Key: " + toleration.Key + " Operator: " + toleration.Operator + " Effect: " + toleration.Effect
 }
 
+func TestFromCrdToAPIJobs(t *testing.T) {
+	jobs := []*rayv1api.RayJob{&JobNewClusterTest}
+	job := FromCrdToAPIJobs(jobs)[0]
```
Comment (dentiny) [inline apiserver/pkg/model/converter_test.go:675]: Could you please help pre-allocate here?
https://github.com/ray-project/kuberay/blob/747708b7cc123eb270ada8f84dbc4a988836db03/apiserver/pkg/model/converter.go#L447

File: apiserver/pkg/model/converter_test.go [hunk 3]
```diff
@@ -568,6 +572,21 @@ func TestAutoscalerOptions(t *testing.T) {
...
 
+func TestNilAutoscalerOptions(t *testing.T) {
+	options := convertAutoscalingOptions(nil)
+	assert.Nil(t, options)
+}
+
+func TestFromCrdToAPIClusters(t *testing.T) {
+	clusters := []*rayv1api.RayCluster{&ClusterSpecTest}
+	clusterEventsMap := map[string][]corev1.Event{}
+	cluster := FromCrdToAPIClusters(clusters, clusterEventsMap)[0]
```
Comment (dentiny) [inline apiserver/pkg/model/converter_test.go:583]: Unfortunately, I think all of your tests need a small modification: you need to check returned result size, then access the first element.

File: apiserver/pkg/model/converter_test.go (hunk 1 above)
Comment (dentiny) [reply apiserver/pkg/model/converter_test.go:580]: Just to clarify, pre-allocation means `apiClusters := make([]*api.Cluster, 0, N) `

**Review by dentiny — APPROVED** [review]
Thanks!

**Review by kevin85421 — APPROVED** [review]
